File: solutions/0007-reverse-integer/solution.c
int reverse(int x){
    int rem,rev=0;
        while (x != 0) {
        rem = x % 10;
        x = x / 10;
        
        // Check for overflow before updating rev
        if (rev > INT_MAX / 10 || (rev == INT_MAX / 10 && rem > INT_MAX % 10)) {
            return 0; // Overflow, return 0
        }
        if (rev < INT_MIN / 10 || (rev == INT_MIN / 10 && rem < INT_MIN % 10)) {
            return 0; // Underflow, return 0
        }
        
        rev = rev * 10 + rem;
    }
    return rev;
}

File: solutions/0007-reverse-integer/solution.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if(x<0):
            if -int(str(x)[1:][::-1])<-2**31:
                return 0
            else:
                return -int(str(x)[1:][::-1])
        if int(str(x)[::-1])>2**31-1:
            return 0
        else:
            return int(str(x)[::-1])
